server/cmd/mmctl/commands/auth.go

```diff
@@ -164,7 +165,11 @@ func loginCmdF(cmd *cobra.Command, args []string) error {
 	allowInsecureSHA1 := viper.GetBool("insecure-sha1-intermediate")
 	allowInsecureTLS := viper.GetBool("insecure-tls-version")
 
-	url := strings.TrimRight(args[0], "/")
+	instanceURL := strings.TrimRight(args[0], "/")
+	_, urlErr := url.ParseRequestURI(instanceURL)
+	if urlErr != nil {
+		return fmt.Errorf("could not parse the instance url: %w", urlErr)
+	}
```

Comment:
Validating the URL is a good starting point. Let's also call the `/api/v4/ping` endpoint to ensure we can reach the server.

Reply:
Okay

Reply:
Hello @hanzei  I can't figure out which client method I can use to make an api call with `url + /api/v4/ping`
What looks similar is `c.GetPing`, `c.GetPingWithOptions` and c.GetPingWithFullServerStatus` but none has a way for me to insert the url. Any idea how I can do this?

Reply:
You need to construct a new `model.Client4` using the URL that the user provided.

Reply:
Okay

Reply:
Hello @hanzei  did you mean `url + /api/v4/system/ping`?